### PR TITLE
Feature: Git Integration Phase 3

### PR DESCRIPTION
**Pull request by ferrariofilippo**
**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Related #11014 

**IMPORTANT**
Please, if you try checking out remotes, place a breakpoint at line 149 of GitHelpers.cs and make sure that the method is executed only once
There was a bug where it would keep on creating new branches because of selection changing. It should be fixed now, but it's better to be careful 

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [x] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [x] Did you implement any design changes to an existing feature?
   - [x] Was this change approved?
- [x] Are there any other steps that were used to validate these changes?
   1. Open app and navigate to a git directory
   2. Click on the branch name
   3. Click on `New Branch`

**Screenshots (optional)**

### CODE REVIEW

File: src/Files.App/Dialogs/AddBranchDialog.xaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+﻿<!--  Copyright (c) 2023 Files Community. Licensed under the MIT License. See the LICENSE.  -->
+<ContentDialog
+	x:Class="Files.App.Dialogs.AddBranchDialog"
+	xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+	xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+	xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+	xmlns:helpers="using:Files.App.Helpers"
+	xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+	x:Name="AddBranch"
+	Title="{helpers:ResourceString Name=NewBranch}"
```
Comment (yaira2) [inline src/Files.App/Dialogs/AddBranchDialog.xaml:10]:
```suggestion
	Title="{helpers:ResourceString Name=CreateBranch}"
```

File: src/Files.App/Dialogs/AddBranchDialog.xaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	BorderThickness="0"
```
Comment (yaira2) [inline src/Files.App/Dialogs/AddBranchDialog.xaml:11]: Do we really need this?

**Comment by hecksmosis** [comment]
I am getting a bug where when you open the branches menu it glitches. 

https://github.com/files-community/Files/assets/55754091/a604de7c-f833-4e25-b936-1962469e7f73

**Comment by ferrariofilippo** [comment]
> I am getting a bug where when you open the branches menu it glitches.

Does it eventually stop flickering? That might be related to the changes being applied, and so the directory info being updated. We might fix that using some kind of `BulkOperation` that we use for other operations

**Comment by hecksmosis** [comment]
> > I am getting a bug where when you open the branches menu it glitches.
> 
> Does it eventually stop flickering? That might be related to the changes being applied, and so the directory info being updated. We might fix that using some kind of `BulkOperation` that we use for other operations

Yes, it stops after a bit. Also, this is not as much of a problem in directories with fewer branches

**Comment by ferrariofilippo** [comment]
@hecksmosis can you please test if the last commit fixes the issue?

**Comment by hecksmosis** [comment]
> @hecksmosis can you please test if the last commit fixes the issue?

I still get the flickering on the latest commit

**Comment by ferrariofilippo** [comment]
I can't reproduce the bug, but I know where it may come from, so I'm going trial and error

**Comment by hecksmosis** [comment]
> I can't reproduce the bug, but I know where it may come from, so I'm going trial and error

Ok, ping me and I'll test

**Comment by ferrariofilippo** [comment]
Can you test now, I pushed a commit that should fix it

**Comment by hecksmosis** [comment]
Yes, it's fixed (btw, it was more like 110 branches so should be robust enough)

**Review by yaira2 — APPROVED** [review]
LGTM

**Review by 0x5bfa — COMMENTED** [review]
Great work!🎉 Thank you @ferrariofilippo!

**Pull request merged**